File: Empresa Adoptfest/proyecto/app/Models/DonacionEspecie.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class DonacionEspecie extends Model
{
    protected $table = 'donaciones_especie';

    protected $fillable = [
        'user_id',
        'categoria',
        'especie_destino',
        'descripcion',
        'cantidad',
        'direccion_recoleccion',
        'telefono_contacto',
        'estado',
        'puntos_otorgados',
        'confirmado_at',
    ];

    protected $casts = [
        'confirmado_at' => 'datetime',
    ];

    const PUNTOS_POR_CATEGORIA = [
        'alimento'      => 50,
        'higiene'       => 30,
        'juguetes'      => 15,
        'cobijas_camas' => 40,
        'medicamentos'  => 60,
        'otros'         => 10,
    ];

    public function user()
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    public function cita()
    {
        return $this->hasOne(CitaDonacion::class, 'donacion_especie_id');
    }

    public function calcularPuntos(): int
    {
        $valorUnidad = self::PUNTOS_POR_CATEGORIA[$this->categoria] ?? 0;
        return $valorUnidad * max(1, $this->cantidad);
    }
}

File: Empresa Adoptfest/proyecto java/backend/src/main/java/com/adoptfest/backend/model/DonacionEspecie.java
package com.adoptfest.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "donaciones_especie")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DonacionEspecie {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CategoriaDonacion categoria;

    @Enumerated(EnumType.STRING)
    @Column(name = "especie_destino", nullable = false)
    private EspecieDestino especieDestino;

    @Column(length = 255)
    private String descripcion;

    @Column(nullable = false)
    private Integer cantidad;

    @Column(name = "direccion_recoleccion", nullable = false)
    private String direccionRecoleccion;

    @Column(name = "telefono_contacto", nullable = false)
    private String telefonoContacto;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private EstadoDonacionEspecie estado = EstadoDonacionEspecie.PENDIENTE;

    @Column(name = "puntos_otorgados")
    private Integer puntosOtorgados;

    @Column(name = "oculto_para_usuario", nullable = false)
    @Builder.Default
    private Boolean ocultoParaUsuario = false;

    @Column(name = "confirmado_at")
    private LocalDateTime confirmadoAt;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    /** Misma fórmula que tenías en Laravel: puntos por unidad × cantidad (mínimo 1). */
    public int calcularPuntos() {
        return categoria.getPuntosPorUnidad() * Math.max(1, cantidad);
    }
}